internal/repository/file.go

```diff
@@ -301,6 +301,15 @@ func (r *FileRepository) Copy(source, destination fyne.URI) error {
 //
 // Since: 2.0
 func (r *FileRepository) Move(source, destination fyne.URI) error {
+	listSrc, _ := r.CanList(source)
+	if listSrc {
+		err := os.Rename(source.Path(), destination.Path())
+		if err == nil {
+			return nil
+		}
+		// fallthrough to slow move
+	}
+
 	// NOTE: as far as I can tell, golang does not have an optimized Move
 	// function - everything I can find on the 'net suggests doing more
 	// or less the equivalent of GenericMove(), hence why that is used.
```

Comment:
Does the above change not invalidate this comment? I think it can be removed. I also kind of don't like that it says "golang" lol so that motivates me more to remove it :)

Reply:
Is it not possible to use the same `os.Rename` function for files?

Reply:
It might be - I was only looking at folders.

Whatever we have to leave the slow path in as well because any Move (/Rename) can fail due to filesystem traversal or cross-volume issues.

Reply:
Alright. Lets add a fast path for regular files as a followup

Reply:
Opened https://github.com/fyne-io/fyne/issues/5517